test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/TestRequestSenderTests.cs

```diff
@@ -470,6 +470,18 @@ public void DiscoverTestShouldNotifyLogMessageIfClientDisconnectedWithClientExit
         _mockDiscoveryEventsHandler.Verify(eh => eh.HandleRawMessage(It.Is<string>(s => !string.IsNullOrEmpty(s) && s.Equals("Serialized Stderr"))), Times.Once);
     }
 
+    [TestMethod]
+    public void DiscoverTestShouldNotifyDiscoveryCompleteIfClientDisconnectedBeforeDiscovery()
```

Comment:
Not sure how hard it is with current test state but it would be interesting to add a test to show we are not calling the callback multiple times.

Reply:
The current test already sets up the verifier with `Times.Once` and that should ensure just that.